File: JS/1-two-sum.js
// map

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let umap = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (umap.has(nums[i])) {
            return [i, umap.get(nums[i])];
        }
        umap.set(target - nums[i], i);
    }
};

File: C++/1-two-sum.cpp
#include "header.h"

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> res;
        for (int i = 0; i < nums.size(); i++) {
            int rest = target - nums[i];
            auto iter = res.find(rest);
            if (iter != res.end()) {
                return {res[rest], i};
            }
            res.insert(std::pair<int, int> (nums[i], i));
        }
        return {};
    }
};